core/object/undo_redo.cpp

```diff
@@ -161,18 +166,23 @@ void UndoRedo::add_undo_method(const Callable &p_callable) {
 		return;
 	}
 
-	Object *object = p_callable.get_object();
-	ERR_FAIL_NULL(object);
+	ObjectID object_id = p_callable.get_object_id();
+	Object *object = ObjectDB::get_instance(object_id);
+	ERR_FAIL_COND(object_id.is_valid() && object == nullptr);
 
 	Operation undo_op;
 	undo_op.callable = p_callable;
-	undo_op.object = p_callable.get_object_id();
-	if (Object::cast_to<RefCounted>(object)) {
+	undo_op.object = object_id;
+	if (object && Object::cast_to<RefCounted>(object)) {
```

Comment:
I've been investigating the performance of `cast_to()` in the case of null objects. My suggestion would be to avoid doing the null check explicitly at the call site and let `cast_to()` do it if desired (it will be inlined). I'm making a PR precisely adding the check so it will be redundant here.

Reply:
Here it is: #82903.